rllib/algorithms/algorithm.py

```diff
@@ -936,10 +939,20 @@ def duration_fn(num_units_done):
             if self.reward_estimators:
                 # Compute off-policy estimates
                 metrics["off_policy_estimator"] = {}
-                total_batch = concat_samples(all_batches)
-                for name, estimator in self.reward_estimators.items():
-                    estimates = estimator.estimate(total_batch)
-                    metrics["off_policy_estimator"][name] = estimates
+                estimates = defaultdict(list)
+                # for each batch run the estimator's fwd pass
+                for batch in all_batches:
+                    for name, estimator in self.reward_estimators.items():
+                        estimate_result = estimator.estimate(
+                            batch, split_by_episode=self.config["ope_split_by_episode"]
```

Comment:
added this optional parameter that is default to True in algorithm config. This will allow bandit users to gain speed up by setting this to False.

Reply:
should we name this ``split_batch_by_episode``?
also a random suggestion, I think we can reduce a level of nesty-ness, since the outmost ``if self.reward_estimators`` is not necessary:

```
estimates = defaultdict(list)
for name, estimator in self.reward_estimators.items():
    for batch in all_batches:
        estimate_result = estimator.estimate(...)
        estimates[name].append(estimate_result)

if estimates:
    metrics["off_policy_estimator"] = {}
for name, estimate_list in estimates.items():
    avg_estimate = tree....
    metrics["off_policy_estimator"][name] = avg_estimate
```

just to make the code look a bit nicer.

Reply:
done.

Reply:
for name: I don't know, I think ope should be part of this since it's an `evaluation()` variable and users need to know from the name that this is for OPE only. So something like `ope_split_batch_by_episode` is kinda the name that I want to put here but it's more verbose which I think is ok?

Reply:
yeah, makes sense, ope_split_batch_by_episode sounds much better.

Reply:
done